app/src/main/java/com/github/sdpcoachme/schedule/CreateEventActivity.kt

```diff
@@ -0,0 +1,609 @@
+        if (email.isEmpty()) {
+            val errorMsg = "New event did not receive an email address.\n Please return to the login page and try again."
+            ErrorHandlerLauncher().launchExtrasErrorHandler(this, errorMsg)
+        } else {
+            setContent {
+                CoachMeTheme {
+                    Surface(color = MaterialTheme.colors.background) {
+                        NewEvent(database)
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+@Composable
+fun NewEvent(database: Database) {
+    val startDateSheet = rememberSheetState()
+    val startTimeSheet = rememberSheetState()
+    val endDateSheet = rememberSheetState()
+    val endTimeSheet = rememberSheetState()
+    val colorSheet = rememberSheetState()
+    val context = LocalContext.current
+    val formatterEventDate = EventOps.getEventDateFormatter()
+    val formatterUserDate = EventOps.getDayFormatter()
+    val formatterUserTime = EventOps.getTimeFormatter()
+
+    var eventName by remember { mutableStateOf("") }
+    var start by remember { mutableStateOf(EventOps.getDefaultEventStart()) }
+    var end by remember { mutableStateOf(EventOps.getDefaultEventEnd()) }
+    var description by remember { mutableStateOf("") }
+    var selectedColor by remember { mutableStateOf(EventColors.DEFAULT.color) }
+
+    Scaffold(
+        modifier = Modifier.testTag(CreateEventActivity.TestTags.SCAFFOLD),
+        topBar = {
+            fun goBackToScheduleActivity() {
+                val intent = Intent(context, ScheduleActivity::class.java)
+                context.startActivity(intent)
+            }
+            val event = Event(eventName, selectedColor.value.toString(), start.format(formatterEventDate), end.format(formatterEventDate), description)
+            TopAppBar(
+                title = {
+                    Text(
+                        text = "New Event",
+                        modifier = Modifier.testTag(CreateEventActivity.TestTags.Texts.ACTIVITY_TITLE)
+                    )
+                },
+                navigationIcon = {
+                    IconButton(
+                        onClick = { goBackToScheduleActivity() },
+                        modifier = Modifier.testTag(CreateEventActivity.TestTags.Clickables.CANCEL),
+                    ) {
+                        Icon(
+                            imageVector = Icons.Filled.Cancel,
+                            contentDescription = "Cancel",
+                            tint = MaterialTheme.colors.onPrimary,
+                            modifier = Modifier.testTag(CreateEventActivity.TestTags.Icons.CANCEL_ICON),
+                        )
+                    }
+                },
+                actions = {
+                    IconButton(
+                        onClick = {
+                            EventOps.addEvent(event, database).thenAccept {
+                                goBackToScheduleActivity()
```

Comment:
Here, before saving the new creating event, we could add a check that entered parameters are valid, ie startDate < endDate. If not we can just show a popup "invalid parameters" and wait for the user to correct them.